### PR TITLE
Initial version

### PR DESCRIPTION
**Pull request by jnoss**

### CODE REVIEW

**Review by tfhartmann — CHANGES_REQUESTED** [review]
Couple of things I think we should add out of the gate - Lets add a CODEOWNERS file and the LICENSE file using the MIT license

**Comment by jnoss** [comment]
@tfhartmann both CODEOWNERS and LICENSE (MIT) are on master - I added them before I started this PR for the initial version of the code

File: s3-dest.tf [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+resource "aws_s3_bucket" "dest" {
+  provider = "aws.dest"
+  bucket   = "${var.dest_bucket_name}"
+  region   = "${var.dest_region}"
+  policy   = "${data.aws_iam_policy_document.dest_bucket_policy.json}"
+
+  versioning {
+    enabled = true
```
Comment (tfhartmann) [inline s3-dest.tf:55]: This might be worth making a parameter, since it's possible that versioning isn't always wanted on source/dest or both.

**Review by tfhartmann — APPROVED** [review]
All in all this looks great! Small comment regarding versioning, but I say :shipit:

File: s3-dest.tf (hunk 1 above)
Comment (jnoss) [reply s3-dest.tf:55]: I thought the same thing myself, but apparently versioning is required to use cross-region replication: https://docs.aws.amazon.com/AmazonS3/latest/dev/crr.html
```
Requirements for cross-region replication:

The source and destination buckets must have versioning enabled. For more information about versioning, see Using Versioning.
```

Comment (tfhartmann) [reply s3-dest.tf:55]: LOL Damn the man!

**Pull request merged**